core/src/main/java/com/squarespace/template/plugins/platform/CommerceFormatters.java

```diff
@@ -466,18 +468,67 @@ private static String defaultSubscriptionPriceString(JsonNode productNode) {
 
       return sb.toString();
     }
+  }
 
-    private static String getMoneyString(JsonNode moneyNode, Context ctx) {
-      if (CommerceUtils.useCLDRMode(ctx)) {
-        Decimal amount = CommerceUtils.getAmountFromMoneyNode(moneyNode);
-        String currencyCode = CommerceUtils.getCurrencyFromMoneyNode(moneyNode);
-        return PluginUtils.formatMoney(amount, currencyCode, ctx.cldr());
-      } else {
-        Decimal legacyAmount = CommerceUtils.getLegacyPriceFromMoneyNode(moneyNode);
-        StringBuilder buf = new StringBuilder();
-        CommerceUtils.writeLegacyMoneyString(legacyAmount, buf);
-        return buf.toString();
+  protected static class SubscriptionPriceFormatter extends BaseFormatter {
+
+    private Instruction template;
+
+    public SubscriptionPriceFormatter() {
+      super("subscription-price", false);
+    }
+
+    @Override
+    public void initialize(Compiler compiler) throws CodeException {
+      String source = loadResource(CommerceFormatters.class, "subscription-price.html");
+      this.template = compiler.compile(source.trim()).code();
+    }
+
+    @Override
+    public void apply(Context ctx, Arguments args, Variables variables) throws CodeExecuteException {
+      Variable var = variables.first();
+      String subscriptionId = args.isEmpty() ? null : args.first();
+      JsonNode node = var.node();
+      StringBuilder buffer = new StringBuilder();
+      ObjectNode subscriptionResults = JsonUtils.createObjectNode();
+
+      JsonNode pricingOptions = CommerceUtils.getPricingOptionsAmongLowestVariant(node);
+
+      if (pricingOptions != null && pricingOptions.size() > 0) {
+        if (CommerceUtils.hasVariants(node)) {
+          JsonNode subscriptionFromPricingNode = CommerceUtils.getSubscriptionMoneyFromPricingOptions(pricingOptions, subscriptionId);
+
+          subscriptionResults.put("fromText", StringUtils.defaultIfEmpty(
+                  ctx.resolve(Constants.PRODUCT_PRICE_FROM_TEXT_KEY).asText(), "from {fromPrice}"));
+          subscriptionResults.put("formattedFromPrice", CommerceUtils.getMoneyString(subscriptionFromPricingNode, ctx));
+        }
+
+        JsonNode firstPricingOption = pricingOptions.get(0);
+
+        if (isOnSale(firstPricingOption)) {
+          subscriptionResults.put("formattedSubscriptionSalePriceText", "{price}");
+          subscriptionResults.put("formattedSubscriptionSalePrice", getSalePriceMoney(firstPricingOption, ctx));
```

Comment:
Just to confirm my understanding: `firstPricingOption` is the cheapest option with the subscription % discount already applied, yes?

Reply:
Chatted offline - all set here! Makes sense given the business requirements.